src/data/finset/intervals.lean

```diff
@@ -82,12 +82,34 @@ lemma union_consecutive {n m l : ℕ} (hnm : n ≤ m) (hml : m ≤ l) :
 by rw [← to_finset, ← to_finset, ← multiset.to_finset_add,
   multiset.Ico.add_consecutive hnm hml, to_finset]
 
+/-- The union of two intervals that overlap each other is also an interval -/
+lemma union_overlapping {n m l k : ℕ} (hlm : l ≤ m) (hnk : n ≤ k) :
```

Comment:
Are you sure you need these two hypotheses? I think this might be true without them.

Reply:
If you don't have them, then you could for example be computing the union of `{1, 2}` and `{5, 6}`, which is not an interval.

Reply:
Ah, for some reason I was thinking about intersections